activestorage/lib/active_storage/attached/many.rb

```diff
@@ -29,7 +29,7 @@ def blobs
     #   document.images.attach([ first_blob, second_blob ])
     def attach(*attachables)
       if record.persisted? && !record.changed?
-        record.public_send("#{name}=", blobs + attachables.flatten)
+        record.public_send("#{name}=", (change&.attachables || blobs) + attachables.flatten)
```

Comment:
This is the same code block as in the `else` clause - can we DRY this up, thanks :+1: